### DISCUSSION TITLE
how to handle "[key: string]: any" combined with specific keys

### DISCUSSION THREAD

**Discussion by aboveyunhai** [open-circle/valibot#400] [Q&A] [answered]
how to handle this typescript case in valibot, I search the doc and didn't find an answer.
```ts
 type Test {
    specific?: string;
    [key: string]: any;
}
```
since `valibot` will strip any additional keys away during `safeParse()`

**Comment by fabian-hiller** [comment, marked as answer]
Thank you for using Valibot! Here is the solution. You can read more about it in the docs: https://valibot.dev/guides/objects/

```ts
import * as v from 'valibot';

const TestSchema = v.object(
  { specific: v.optional(v.string()) },
  v.any()
);
```

**Reply by fabian-hiller** [reply]
Hint: The schema function `any` exists only for completeness and is not recommended in practice. Instead, [`unknown`](https://valibot.dev/api/unknown/) should be used to accept unknown data.

**Reply by aboveyunhai** [reply]
@fabian-hiller  thx. was about to ask the question, will do the `unknown` for full type safety, as long as the additional field remain after parse.
